src/Mvc/Mvc.ViewFeatures/src/RazorComponents/HtmlRenderer.cs

```diff
@@ -228,6 +228,12 @@ private int RenderChildren(HtmlRenderingContext context, ArrayRange<RenderTreeFr
         {
             var candidateIndex = position + i;
             ref var frame = ref frames.Array[candidateIndex];
+
+            if (frame.FrameType == RenderTreeFrameType.ElementReferenceCapture)
```

Comment:
Could we move this check inside the branch below? That is, we only need to check if it's `ElementReferenceCapture` once we've already observed that it's *not* `RenderTreeFrameType.Attribute`. Then no extra checks will occur on most cycles.